packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -278,6 +279,9 @@ class EditableTextState extends State<EditableText> implements TextInputClient {
   final LayerLink _layerLink = new LayerLink();
   bool _didAutoFocus = false;
 
+  @override
+  bool get wantKeepAlive => widget.focusNode.hasFocus;
```

Comment:
The focusNode seems like another common cause where the mixin could do all the fiddly work for you.